### ISSUE TITLE
random wallet_basic failure

### ISSUE THREAD

**Issue by instagibbs** [bitcoin/bitcoin#19853]
https://travis-ci.org/github/bitcoin/bitcoin/jobs/722945532#L2583

Probably can be avoided fairly easily.

**Comment by MarcoFalke** [comment]
I don't see any failure in the link. See https://github.com/bitcoin/bitcoin/issues/15847#issuecomment-487591130

**Comment by Sjors** [comment]
Here's one from #18788: [log](https://travis-ci.org/github/bitcoin/bitcoin/jobs/722952265#L2561):

```
020-09-01T05:18:47.709000Z TestFramework (INFO): Initializing test directory /Users/travis/build/bitcoin/bitcoin/ci/scratch/test_runner/test_runner_₿_🏃_20200901_051608/wallet_basic_178
...
25662020-09-01T05:19:50.052000Z TestFramework (INFO): check -reindex
25672020-09-01T05:20:03.516000Z TestFramework (ERROR): Assertion failed
2568Traceback (most recent call last):
...
2575test_framework.authproxy.JSONRPCException: Amount out of range (-3)
2576
2577During handling of the above exception, another exception occurred:
2578
2579Traceback (most recent call last):
2580  File "/Users/travis/build/bitcoin/bitcoin/ci/scratch/build/bitcoin-x86_64-apple-darwin16/test/functional/test_framework/test_framework.py", line 118, in main
2581    self.run_test()
2582  File "/Users/travis/build/bitcoin/bitcoin/ci/scratch/build/bitcoin-x86_64-apple-darwin16/test/functional/wallet_basic.py", line 601, in run_test
2583    assert_raises_rpc_error(-6, "Transaction has too long of a mempool chain", self.nodes[0].sendtoaddress, sending_addr, node0_balance - Decimal('0.01'))
```

**Comment by instagibbs** [comment]
Yep same log. Sorry for blowing the other one away!

**Comment by Sjors** [comment]
Here's one that's still in tact: https://travis-ci.org/github/bitcoin/bitcoin/jobs/722800245#L6406

**Comment by practicalswift** [comment]
Another case of `test_framework.authproxy.JSONRPCException: Amount out of range (-3)`:

* https://travis-ci.org/github/bitcoin/bitcoin/jobs/723023488#L2575

Context:

```
test_framework.authproxy.JSONRPCException: Amount out of range (-3)
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/Users/travis/build/bitcoin/bitcoin/ci/scratch/build/bitcoin-x86_64-apple-darwin16/test/functional/test_framework/test_framework.py", line 118, in main
    self.run_test()
  File "/Users/travis/build/bitcoin/bitcoin/ci/scratch/build/bitcoin-x86_64-apple-darwin16/test/functional/wallet_basic.py", line 601, in run_test
    assert_raises_rpc_error(-6, "Transaction has too long of a mempool chain", self.nodes[0].sendtoaddress, sending_addr, node0_balance - Decimal('0.01'))
```

**Comment by n-thumann** [comment]
Same error when testing wallet_basic here: https://travis-ci.org/github/bitcoin/bitcoin/jobs/723365668#L2572 for https://github.com/bitcoin/bitcoin/pull/18309. Would someone mind to restart the job or is there something else I can do? :)

**Comment by fjahr** [comment]
Potential fix: #19876, looking for feedback :)
